src/net/client.py

```diff
@@ -15,18 +15,22 @@
 
 from src.constants import TxStatus
 
-dns = "alpha4.starknet.io"
+T = TypeVar("T")
 
 
 class Client:
-    def __init__(self, retry_config: Optional[RetryConfig] = None):
+    @classmethod
```

Comment:
I think we'll need to replace it with a staticmethod anyway. Singer accepts additional arguments that we will need to provide.